update: update colorizer docs to reflect new over/under functionallity (#38)
geo-engine/docs: src/datatypes/colorizer.md

Different variants of colorizers perform different kinds of mapping.
In general, there are two families of colorizers: **_gradient_** and **_palette_**.
_Gradients_ are used to interpolate a continuous spectrum of colors between explicitly stated tuples (`breakpoints`) of a value and a color.
A `palette` colorizer on the other hand, is used to generate a discrete set of colors, each mapped to a specific value.

There are three miscellaneous fields in both of the gradient colorizers, namely `noDataColor`, `overColor` and `underColor`.
The field `noDataColor` is used for all missing, `NaN` or no data values.
The fields `overColor` and `underColor` are used for all overflowing values.
For instance, if there are breakpoints defined from `0` to `10`, but a value of `-5` or `11` is mapped to a color, the respective field will be chosen instead.
This way, you can specifically highlight values that lie outside of a given range.

For a `palette` colorizer, there are no `overColor` and `underColor` fields.
If a given value does not match any entry in the palette's definition, it is mapped to the `defaultColor`.

> The `noDataColor` field is used for values, that cannot be mapped, for instance `NaN`.

Reviewer: ```suggestion
The `noDataColor` field is used for no data values or `NaN`.
```

But somehow this is already described in line 10.
So you could as well remove it.

Author: 994b87e

A colorizer specifies a mapping between values and pixels/objects of an output image.
Different variants of colorizers perform different kinds of mapping.
In general, there are two families of colorizers: **_gradient_** and **_palette_**.
_Gradients_ are used to interpolate a continuous spectrum of colors between explicitly stated tuples (`breakpoints`) of a value and a color.
A `palette` colorizer on the other hand, is used to generate a discrete set of colors, each mapped to a specific value.

There are three miscellaneous fields in both of the gradient colorizers, namely `noDataColor`, `overColor` and `underColor`.
The field `noDataColor` is used for all missing, `NaN` or no data values.
The fields `overColor` and `underColor` are used for all overflowing values.
For instance, if there are breakpoints defined from `0` to `10`, but a value of `-5` or `11` is mapped to a color, the respective field will be chosen instead.
This way, you can specifically highlight values that lie outside of a given range.

For a `palette` colorizer, there are no `overColor` and `underColor` fields.

> Instead, a `noDataColor` is used as a fallback and `defaultColor` is mapped to any value not explicitly declared in the palette's definition.

Reviewer: "Instead" makes no sense with respect to no data, since this is used for no data values and not as a fallback.

The default color is the fallback for all values that do not match any specified value.

Author: [1ee490a](https://github.com/geo-engine/docs/commit/1ee490a4e4b6705aec3bf9ce98ba5996159d2d97#diff-5bf7663539adc56559802f7881e6c9cc03912949d6a679e8ebd4227da9e3be3eR16)

A colorizer specifies a mapping between values and pixels/objects of an output image.
Different variants of colorizers perform different kinds of mapping.

> Usually, there are three miscellaneous fields in each colorizer, namely `noDataColor`, `overColor` and `underColor`.

Reviewer: It is not that easy anymore.

It should be something like this.

There are colors for values or value ranges, then a color for NO DATA, and then color(s) for data that is out of range, e.g. over and under colors for gradients or a default color for palettes.

Author: 61fc7724b85e21f45ba69bd81c464b7010e4a8ac

Ok, I've rewritten it much more deliberately now.

  "type": "linearGradient",
  "breakpoints": [
    {
      "value": 0.0,
      "color": [204, 229, 255, 255]
    },
    {
      "value": 99.99,
      "color": [0, 0, 255, 255]
    }
  ],
  "noDataColor": [0, 0, 0, 0],
  "overColor": [224, 224, 224, 255],
  "underColor": [255, 255, 255, 255]
}
```

## Logarithmic Gradient

A logarithmic gradient logarithmically interpolates values within breakpoints of a color table and allows only positive values.
This colorizer is particularly useful in situations,

> where your data grows rapidly and minor changes in the lower numbers wouldn't be recognizable anymore.

Reviewer: ```suggestion
where the data values increase exponentially and minor changes in the lower numbers would not be recognizable anymore.
```

Author: 1ee490a

    }
  ],
  "noDataColor": [0, 0, 0, 0],
  "overColor": [255, 0, 0, 255],

>   "underColor": [0, 0, 255, 255]

Reviewer: Are these reasonable over and under colors?

You have a gradient from white to black and then suddenly it is blue? You could continue with white or black, for instance.

    }
  ],
  "noDataColor": [0, 0, 0, 0],
  "overColor": [255, 0, 0, 255],

>   "underColor": [0, 0, 255, 255]

Reviewer: Same here.

Maybe you should update the description above. You could have a focus on a certain area, e.g., to highlight differences in temperature, but you declear everything above 20° just as hot and everything below 0 just as cold. But you can be very distinctive between 0 and 20°.

Author: I tried to come up with a reasonable setting now and hopefully more representative colors.

      "color": [255, 255, 255, 255]
    },
    {
      "value": 100.0,
      "color": [0, 0, 0, 255]
    }
  ],
  "noDataColor": [0, 0, 0, 0],
  "overColor": [255, 255, 255, 255],

>   "underColor": [0, 0, 0, 255]

Reviewer: Shouldn't this be the other way around?

Author: 2744112

correct

}
```

## Palette

> A palette maps values as classes to a certain color. Unmapped values result in the `defaultColor`.

Reviewer: ```suggestion
A palette maps values as classes to a certain color.
Unmapped values result in the `defaultColor`.
```

Please use one sentence per line.

Author: 7f6e236

Whoops
